src/OpenSpaceToolkit/Astrodynamics/Trajectory/State.cpp

```diff
@@ -225,16 +215,16 @@ void State::print(std::ostream& anOutputStream, bool displayDecorator) const
     ostk::core::utils::Print::Line(anOutputStream)
         << "Instant:" << (this->instant_.isDefined() ? this->instant_.toString() : "Undefined");
     ostk::core::utils::Print::Line(anOutputStream)
-        << "Position:" << (this->position_.isDefined() ? this->position_.toString(12) : "Undefined");
+        << "Position:" << (this->isDefined() ? this->accessPosition().toString(12) : "Undefined");
     ostk::core::utils::Print::Line(anOutputStream)
-        << "Velocity:" << (this->velocity_.isDefined() ? this->velocity_.toString(12) : "Undefined");
+        << "Velocity:" << (this->isDefined() ? this->accessVelocity().toString(12) : "Undefined");
 
     displayDecorator ? ostk::core::utils::Print::Footer(anOutputStream) : void();
 }
 
 State State::Undefined()
 {
-    return {Instant::Undefined(), Position::Undefined(), Velocity::Undefined()};
+    return {Instant::Undefined(), VectorXd(6), Frame::Undefined()};
```

Comment:
I did it so it's tight to pos-vel as constructor will throw an error if size != 6, but it will eventually get generalised so I implemented your suggestion already.